### PR TITLE
chore(pod): remove classic theme - FE-3118

### PR DESCRIPTION
**Pull request by mkrds**
### Proposed behaviour
This PR removes classic theme support from `Pod` component and refactors old stories to use new CSF/MDX format.

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### CODE REVIEW

File: src/components/pod/pod.stories.js [hunk 1]
```diff
@@ -1,100 +1,58 @@
...
-    );
-  };
-
-  const metadata = {
-    themeSelector,
+export default {
+  title: 'Pod/Test',
+  component: Pod,
+  parameters: {
+    themeSelector: dlsThemeSelector,
```
Comment (edleeks87) [inline src/components/pod/pod.stories.js:12]: Is a `themeSelector` needed for this story?

File: src/components/pod/pod.stories.mdx [hunk 2]
```diff
@@ -4,16 +4,236 @@ import Heading from '../heading';
...
+# Pod
+Presents content grouped visually together. This can be text, or other components. A good example is a ‘tile’ showing contact information for an individual.
+
+Configure Pod and Fieldset components to work together, or choose the pre-configured Show/Edit Pod component:
+
+The ShowEditPod component automatically presents content as a read-only Pod, until the user clicks an edit icon, which shows the same information in an editable Fieldset.
+
+But, configuring Pod and Fieldset components manually will give you more customization options, like the following.
+
+The On Edit property shows a standard edit icon, which can be used to show a fieldset. You can choose whether this icon appears inside or outside the pod, and whether it appears only on hover. You can also choose whether clicking only the icon triggers the On Edit property, or clicking anywhere on the pod.
```
Comment (edleeks87) [inline src/components/pod/pod.stories.mdx:20]:
```suggestion
The onEdit property shows a standard edit icon, which can be used to show a fieldset. You can choose whether this icon appears inside or outside the pod, and whether it appears only on hover. You can also choose whether clicking only the icon triggers the onEdit property, or clicking anywhere on the pod.
```

File: src/components/pod/pod.stories.js (hunk 1 above)
Comment (mkrds) [reply src/components/pod/pod.stories.js:12]: Don't we provide it by default for all the stories?

File: src/components/pod/pod.stories.mdx [hunk 3]
```diff
@@ -4,16 +4,236 @@ import Heading from '../heading';
...
 />
 
+# Pod
+Presents content grouped visually together. This can be text, or other components. A good example is a ‘tile’ showing contact information for an individual.
+
+Configure Pod and Fieldset components to work together, or choose the pre-configured Show/Edit Pod component:
+
+The ShowEditPod component automatically presents content as a read-only Pod, until the user clicks an edit icon, which shows the same information in an editable Fieldset.
+
+But, configuring Pod and Fieldset components manually will give you more customization options, like the following.
```
Comment (ianforster-sage) [inline src/components/pod/pod.stories.mdx:18]: Starting a sentence with a conjunction like 'but' is usually discouraged in English, try 'however' instead.
```suggestion
The ShowEditPod component automatically presents content as a read-only Pod, until the user clicks an edit icon, which shows the same information in an editable Fieldset. However, configuring Pod and Fieldset components manually will give you more customisation options, like the following:
```

**Review by ianforster-sage — COMMENTED** [review]
Is it atomic to have changes to Grid, Pod and ShowEditPod in the same commit?

**Review by edleeks87 — DISMISSED** [review]

**Review by edleeks87 — DISMISSED** [review]

**Review by ianforster-sage — DISMISSED** [review]

File: cypress/fixtures/commonComponents/pod.json [hunk 4]
```diff
@@ -26,20 +26,20 @@
   "paddingExtraLarge": {
     "padding": "extra-large"
   },
-  "podTypePrimary": {
-    "podType": "primary"
+  "variantPrimary": {
```
Comment (DlgSHi) [inline cypress/fixtures/commonComponents/pod.json:29]: can You please also clear up this file after deleting tests from `pod.feature` file?

**Review by DlgSHi — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by ianforster-sage — APPROVED** [review]

**Pull request merged**